examples/python/monitoring/07_hierarchical_agents_monitoring.py

```
@@ -0,0 +1,263 @@
+    print("\n  🔄 Coordination Functions:")
+    for func in coordination_functions:
+        data = stats[func]
+        avg_time = data['total_time'] / data['call_count'] if data['call_count'] > 0 else 0
+        print(f"    {func}: {data['call_count']} calls, {avg_time:.3f}s avg")
+    
+    # Hierarchy efficiency metrics
+    total_management_time = sum(stats[f]['total_time'] for f in management_functions)
+    total_worker_time = sum(stats[f]['total_time'] for f in worker_functions)
+    total_coordination_time = sum(stats[f]['total_time'] for f in coordination_functions)
+    
+    print(f"\n⚖️ Hierarchy Efficiency Metrics:")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)